src/cursor/abstract_cursor.ts

```diff
@@ -680,88 +680,107 @@ export abstract class AbstractCursor<
   }
 }
 
-function nextDocument<T>(cursor: AbstractCursor<T>): T | null {
-  const doc = cursor[kDocuments].shift();
-
-  if (doc && cursor[kTransform]) {
-    return cursor[kTransform](doc) as T;
-  }
-
-  return doc;
-}
-
-const nextAsync = promisify(
-  next as <T>(
-    cursor: AbstractCursor<T>,
-    blocking: boolean,
-    callback: (e: Error, r: T | null) => void
-  ) => void
-);
-
 /**
  * @param cursor - the cursor on which to call `next`
  * @param blocking - a boolean indicating whether or not the cursor should `block` until data
  *     is available.  Generally, this flag is set to `false` because if the getMore returns no documents,
  *     the cursor has been exhausted.  In certain scenarios (ChangeStreams, tailable await cursors and
  *     `tryNext`, for example) blocking is necessary because a getMore returning no documents does
  *     not indicate the end of the cursor.
- * @param callback - callback to return the result to the caller
- * @returns
+ * @param transform - if true, the cursor's transform function is applied to the result document (if the transform exists)
+ * @returns the next document in the cursor, or `null`.  When `blocking` is `true`, a `null` document means
+ * the cursor has been exhausted.  Otherwise, it means that there is no document available in the cursor's buffer.
  */
-export function next<T>(
+async function next<T>(
   cursor: AbstractCursor<T>,
   blocking: boolean,
-  callback: Callback<T | null>
-): void {
+  transform = true
```

Comment:
I assume we are defaulting to `true` here as more methods would transform vs. methods that don't (`hasNext`)?

Reply:
Yeah, that was my logic.  I also considered two other options:

- providing no default
- combining `blocking` and `transform` into an options object and requiring all callsites to explicitly set them for clarity (i.e., `next(this, { blocking: true, transform: false })`

Reply:
I lean towards providing no default, for internal I prefer to make each call site define its inputs and because this isn't saving us a lot of refactors. I don't feel strongly though, feel free to resolve if we agree to leave it as is.